openedx/core/djangoapps/oauth_dispatch/jwt.py

```diff
@@ -289,7 +289,7 @@ def _encode_and_sign(payload, use_asymmetric_key, secret):
         algorithm = settings.JWT_AUTH['JWT_ALGORITHM']
 
     jwk = PyJWK(key, algorithm)
-    return jwt.encode(payload, jwk.key, algorithm=algorithm)
+    return jwt.encode(payload, jwk.key, algorithm=algorithm, headers={'kid': jwk.key_id})
```

Comment:
Thanks.
1. We only want this for the `use_asymmetric_key` case. Hopefully `headers=None` will work.
2. The safest would be to create a [temporary rollout SettingToggle](https://edx.readthedocs.io/projects/edx-toggles/en/latest/how_to/implement_the_right_toggle_type.html#django-setting-toggles). It should be documented with annotations ([see separate documenting toggles doc](https://edx.readthedocs.io/projects/edx-toggles/en/latest/how_to/documenting_new_feature_toggles.html)).
3. Clearly we'll want unit tests once this is ready.